src/Build/BackEnd/TaskExecutionHost/TaskExecutionHost.cs

```diff
@@ -333,6 +333,7 @@ bool ITaskExecutionHost.SetTaskParameters(IDictionary<string, (string, ElementLo
             // "required" so that we can keep track of whether or not they all get set.
             var setParameters = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
             IDictionary<string, string> requiredParameters = GetNamesOfPropertiesWithRequiredAttribute();
+            IDictionary<string, string> allowEmptyStringParameters = GetNamesOfPropertiesWithAllowEmptyStringAttribute();
```

Comment:
I know requiredParameters just above is an IDictionary<string, string>, but I don't see anywhere its values can ever be anything other than string.Empty. Why is this an IDictionary?

To be honest, I'm wondering if you could get by without even a Set. It looks like you have (taskName, location) and TaskPropertyInfo, and you just need to figure out which corresponds to which, and you can avoid allocating anything. I'd guess they're in the same order, but that'd be something to check.

Reply:
Yeah, it is not necessary to be an IDictionary.

Reply:
> To be honest, I'm wondering if you could get by without even a Set
Do you mean without  SetTaskItemParameter(parameter, new TaskItem(parameterValue, parameterLocation.LocationString, allowEmptyString)) ?

Reply:
If you don't need an IDictionary, the logical next option would be to change it to a Set like a HashSet or something, but I'm wondering if you can get by without having a data structure there at all and just accessing the allowEmptyString attribute as needed.

Reply:
Sure, I cann get by without having a data structure there at all.  But I'm think if we should cache this property just like required, output ...